Production/plugins/TrainTupleProducer.cc

```diff
@@ -0,0 +1,694 @@
+
+        edm::Handle<HFRecHitCollection> hf;
+        event.getByToken(hf_token, hf);
+
+        edm::Handle<CaloTowerCollection> caloTowers;
+        event.getByToken(caloTowers_token, caloTowers);
+
+        edm::Handle<reco::TrackCollection> Tracks;
+        event.getByToken(Tracks_token, Tracks);
+
+        edm::Handle<reco::TrackCollection> pataTracks;
+        event.getByToken(pataTracks_token, pataTracks);
+
+        edm::Handle<reco::CaloJetCollection> caloTaus;
+        event.getByToken(caloTaus_token, caloTaus);
+        pat::JetCollection jets;
+
+        edm::ESHandle<CaloGeometry> Geometry = eventsetup.getHandle(Geometry_token);
+
+        caloRecHitCollections AllCaloRecHits;
+        AllCaloRecHits.hbhe= &*hbhe;
+        AllCaloRecHits.ho= &*ho;
+        AllCaloRecHits.hf= &*hf;
+        AllCaloRecHits.eb= &*ebHandle;
+        AllCaloRecHits.ee= &*eeHandle;
+        AllCaloRecHits.Geometry = &*Geometry;
+
+        edm::Handle<std::vector<reco::GenParticle>> hGenParticles;
+        if(isMC){
+            event.getByToken(genParticles_token, hGenParticles);
+            std::vector<reco_tau::gen_truth::GenLepton> GenLeptons = reco_tau::gen_truth::GenLepton::fromGenParticleCollection(*hGenParticles);
+            FillGenLepton(GenLeptons);
+          }
+
+        //check inputs
+        FillL1Objects(*l1Taus, l1_handle_map);
+        FillTracks(*Tracks, "track", *vertices);
+        FillTracks(*pataTracks, "patatrack", *pataVertices);
+        FillVertices(*vertices, "vert");
+        FillVertices(*pataVertices, "patavert");
+        FillCaloRecHit(AllCaloRecHits);
+        FillCaloTowers(*caloTowers);
+        FillCaloTaus(*caloTaus);
+        FillHLTResults(*Trigger_Results, *pSet, trigger_names.triggerNames());
+        auxTuple.Fill();
+        trainTuple.Fill();
+    }
+
+    virtual void endJob() override
+    {
+        TrainTupleProducerData::ReleaseGlobalData();
+    }
+
+private:
+
+  void FillHLTResults(const edm::TriggerResults& Trigger_Results, const edm::ParameterSet& pSet, const std::vector<std::string>& trigger_names){
+    // 1 trovare la posizioe del nome del trigger nel vettore - per ora solo _v4
+    const int position_of_path  = std::distance(trigger_names.begin(), std::find( trigger_names.begin(), trigger_names.end(), "HLT_DoubleMediumChargedIsoPFTauHPS35_Trk1_eta2p1_Reg_v4" )) ;
+    // 2 prendi da trigger results tramite la funzione at l'indice del TriggerResults
+    const edm::HLTPathStatus& hlt_path_status = Trigger_Results.at(position_of_path);
+    // 3 l'indice mi ritorna hltpathstatus che salvo , mi ritorna indice del modulo che ha preso la decisione (ultimo) e state numero che va da 0 a 3
+    const std::vector<std::string>& module_names = pSet.getParameter<std::vector<std::string>>("HLT_DoubleMediumChargedIsoPFTauHPS35_Trk1_eta2p1_Reg_v4"); // ->  dopo devo fare module_names.at(hltpathstatus.index()), questo sara il nome
+    auxTuple().module_names = module_names;
```

Comment:
- module names should be filled only once
- pass the name of the default diTau path as a parameter